### PR TITLE
Add wfmlabs.org to Wikipedia mdfp group. Add wikipedia.org to cookieblocklist.  Update mdfp docs, add tests.

### LINKED ISSUE #1178: en.wikipedia.org breaks page
https://tools.wmflabs.org/pageviews/?project=en.wikipedia.org&platform=all-access&agent=user&range=latest-20&pages=

`en.wikipedia.org` breaks page.

### PR DESCRIPTION
**Pull request by cowlicks**
Closes #1178 

This adds wfmlabs.org to the  Wikipedia Multi-Domain First Parties (MDFP) group. It is under the wikimedia umbrella.

wikipedia.org is added to the cookieblock list. It has an api that some third party sites use, like wikiwand.

I also updated the docs in `multiDomainFirstParties.js`. In other places we expect the domains to be etld+1, but this wasn't documented anywhere, so I note that (related to #1515).

This code was entirely untested before. However it also didn't expose parts that made it easy to test. So I did a small refactoring to make it easier to test. And I also added tests.

I also noticed that `isMultiDomainFirstParty` was O(n) in the number of domains associated with a first party. This is called several times in performance critical code. So I added a small change to make it constant time (related to #1371).

### CODE REVIEW

File: doc/sample_cookieblocklist.txt [hunk 1]
```diff
@@ -554,6 +554,7 @@ where.com
 widgetserver.com
 wikidata.org
 wikimedia.org
+wikipedia.org
```
Comment (ghostwords) [inline doc/sample_cookieblocklist.txt]: Could we break this change out into its own pull request please? Here are some test sites:
- https://www.wikiwand.com/en/Enzo_Ferrari_(automobile) (from #1178)
- http://kevan.org/wikitext/
- https://www.wikiwand.com/en/Comparison_between_U.S._states_and_countries_by_GDP_(nominal)
- https://xefer.com/wikipedia

File: src/js/multiDomainFirstParties.js [hunk 2]
```diff
@@ -181,7 +168,7 @@ var _multiDomainFirstPartiesArray = [
   ["walmart.com", "wal.co"],
   ["wellsfargo.com", "wf.com"],
   ["wikia.com", "wikia.net", "nocookie.net"],
-  ["wikipedia.org", "wikimedia.org", "wikimediafoundation.org", "wiktionary.org",
+  ["wfmlabs.org","wikipedia.org", "wikimedia.org", "wikimediafoundation.org", "wiktionary.org",
```
Comment (ghostwords) [inline src/js/multiDomainFirstParties.js]: Typo: should be `wmflabs.org`!

Nit: missing a separating space. Plus, could we move it to the back of the list? We should have the most important sites up front probably.

File: src/tests/multiDomainFirstParties.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+(function(){
```
Comment (ghostwords) [inline src/tests/multiDomainFirstParties.js]: We actually do already have tests for MDFP, or at least a test, in [tests/utils.js](https://github.com/EFForg/privacybadger/blob/d217a28ee3188d51797f909763ae7d108a1ee63c/src/tests/tests/utils.js#L95-L108).

File: src/tests/multiDomainFirstParties.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  QUnit.module("multiDomainFirstParties.js");
+
+  let mdfp = require('multiDomainFP');
+
+  QUnit.test('isMultiDomainFirstParty test', function (assert) {
+    let testData = [['foo.bar', 'yep.com', 'maybe.idk']],
+      isMdfp = mdfp.makeIsMultiDomainFirstParty(mdfp.makeDomainLookup(testData));
+
+    assert.ok(isMdfp('yep.com', 'maybe.idk') === true);
```
Comment (ghostwords) [inline src/tests/multiDomainFirstParties.js]: No need to compare with `true` or `false`.

File: src/js/multiDomainFirstParties.js [hunk 5]
```diff
@@ -200,40 +187,36 @@ var _multiDomainFirstPartiesArray = [
...
  */
-function _makeDomainLookup() {
-  var arr = _multiDomainFirstPartiesArray;
-  var out = {};
-  var arrLength = arr.length;
-  for (var i = 0; i < arrLength; i++) {
-    var inLength = arr[i].length;
-    for (var j = 0; j < inLength; j++) {
-      out[arr[i][j]] = arr[i];
+function makeDomainLookup(mdfpArray) {
```
Comment (ghostwords) [inline src/js/multiDomainFirstParties.js]: Could we please move the performance improvement into its own PR?

File: src/js/multiDomainFirstParties.js (hunk 2 above)
Comment (ghostwords) [inline src/js/multiDomainFirstParties.js]: Here are some test cases:
- https://tools.wmflabs.org/pageviews/?project=en.wikipedia.org&platform=all-access&agent=user&range=latest-20&pages=
- https://tools.wmflabs.org/geohack/geohack.php?pagename=Panmunjom&params=37.956_N_126.677_E
- https://petscan.wmflabs.org/?depth=10&templates_yes=asbox&sortby=title

**Comment by cowlicks** [comment]
Can you please elaborate on why this should be broken up into separate pull requests?

I'd prefer to keep it together for two reasons:

While I agree that the changes can be done independently, but they are loosely related, and relatively small low-risk.

Splitting this into multiple PR's takes time. It also requires me to track more independent changes while I already have *many* open pull requests. So if I want to build on something here, like add another test to mdfp or something, things start to get complicated like in #1525 and #1524

**Comment by ghostwords** [comment]
It's harder to reason about, review and track changesets the larger they are. When a changeset consists of several, loosely-related changesets, we should break it up. There are at least three separate things going on in this ticket right now, each with its own checklist of things to go over. Could we try it this way (decouple anything that can be decoupled) and see how that goes?

**Comment by ghostwords** [comment]
There are several threads going on this ticket already, besides my requests to split up the PR, and I haven't even started looking at the performance optimization.

**Comment by ghostwords** [comment]
Are we going to include `wikipedia.org` in next week's yellowlist update? Could we have a PR for that please?
